org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockStringNexusScannable.java

```diff
@@ -0,0 +1,90 @@
+package org.eclipse.scanning.example.scannable;
+
+import java.text.MessageFormat;
+
+import org.eclipse.dawnsci.nexus.INexusDevice;
+import org.eclipse.dawnsci.nexus.NXobject;
+import org.eclipse.dawnsci.nexus.NXpositioner;
+import org.eclipse.dawnsci.nexus.NexusException;
+import org.eclipse.dawnsci.nexus.NexusNodeFactory;
+import org.eclipse.dawnsci.nexus.NexusScanInfo;
+import org.eclipse.dawnsci.nexus.NexusScanInfo.ScanRole;
+import org.eclipse.dawnsci.nexus.builder.NexusObjectProvider;
+import org.eclipse.dawnsci.nexus.builder.NexusObjectWrapper;
+import org.eclipse.january.dataset.Dataset;
+import org.eclipse.january.dataset.DatasetFactory;
+import org.eclipse.january.dataset.ILazyWriteableDataset;
+import org.eclipse.january.dataset.SliceND;
+import org.eclipse.scanning.api.IScanAttributeContainer;
+import org.eclipse.scanning.api.points.IPosition;
+import org.eclipse.scanning.api.scan.rank.IScanRankService;
+import org.eclipse.scanning.api.scan.rank.IScanSlice;
+
+public class MockStringNexusScannable extends MockStringScannable implements INexusDevice<NXpositioner> {
+
+	private ILazyWriteableDataset lzValue;
+	
+	public MockStringNexusScannable(String name, String pos, String... permittedValues) {
+		super(name, pos, permittedValues);
+	}
+
+	@Override
+	public NexusObjectProvider<NXpositioner> getNexusProvider(NexusScanInfo info) throws NexusException {
+		final NXpositioner positioner = NexusNodeFactory.createNXpositioner();
+		positioner.setNameScalar(getName());
+		
+		if (info.getScanRole(getName()) == ScanRole.METADATA) {
+			try {
+				// note: assume this scannable is a monitor, so no set value dataset created 
+				positioner.setValue(DatasetFactory.createFromObject(getPosition()));
+			} catch (Exception e) {
+				throw new NexusException("Could not get value for scannable " + getName(), e);
+			}
+		} else {
+			lzValue = positioner.initializeLazyDataset(NXpositioner.NX_VALUE, info.getRank(), String.class);
+			lzValue.setChunking(info.createChunk(false, 8));
+			lzValue.setWritingAsync(true);
+		}
+		
+		registerAttributes(positioner, this);
+		
+		return new NexusObjectWrapper<>(getName(), positioner, NXpositioner.NX_VALUE);
+	}
+	
+	public void setPosition(String value, IPosition position) throws Exception {
+		if (position != null) {
+			write(value, getPosition(), position);
+		}
+	}
+	
+	private void write(String demand, String actual, IPosition pos) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "write" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "demand". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)